src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/SqlConnection.cs

```diff
@@ -53,7 +53,7 @@ public static SqlConnection Create(IPersistentStorageFaultInjector faultInjector
 
             // Enable shared cache so that multiple connections inside of same process share cache
             // see https://sqlite.org/threadsafe.html for more detail
-            var flags = OpenFlags.SQLITE_OPEN_CREATE | OpenFlags.SQLITE_OPEN_READWRITE | OpenFlags.SQLITE_OPEN_SHAREDCACHE;
+            var flags = OpenFlags.SQLITE_OPEN_CREATE | OpenFlags.SQLITE_OPEN_READWRITE | OpenFlags.SQLITE_OPEN_NOMUTEX | OpenFlags.SQLITE_OPEN_SHAREDCACHE;
```

Comment:
Why this flag?

Reply:
This flag says we are only using connection instances on one thread at a time. We were already doing this, but since we didn't specify the flag the database fell back to SQLITE_OPEN_FULLMUTEX which added a second layer of locks.